ddtrace/tracer/span.go

```diff
@@ -464,6 +472,27 @@ func (s *span) setMetric(key string, v float64) {
 	}
 }
 
+// AddSpanLinks appends the given links to the span's span links.
+func (s *span) AddSpanLinks(spanLinks ...ddtrace.SpanLink) {
```

Comment:
I think AddSpanLinks (slice) is riskier than a method like AddSpanLink (singular). Because slices are shared by reference, any modification to the spanLinks slice that the customer could make after calling this method will impact the spanLinks on our span. Either you make a copy of spanLinks in this existing method before putting it on the span (seems like an annoying perf impact), or you accept only a single span link (which is pass by value). 

If you don't need a method for appending multiple span links, you could just change it to AddSpanLink. Alternatively, if you really need the method as-is, I suppose you could document the behavior and put the onus on the customer, e.g: "Callers of AddSpanLinks should avoid modifying the spanLinks slice after passing it in"